iothub/device/src/Microsoft.Azure.Devices.Client.csproj

```diff
@@ -55,13 +55,9 @@
     </None>
   </ItemGroup>
 
-  <!--Project reference dependencies-->
-  <ItemGroup>
-    <ProjectReference Include="..\..\service\src\Microsoft.Azure.Devices.csproj" />
```

Comment:
Why did we even have this dependency to begin with?

Reply:
Seems that we were fixing some merge conflict [errors](https://github.com/Azure/azure-iot-sdk-csharp/commit/cb85816a2fdad9dce0ec2c528ea16772b1702917#diff-1667ae85c249af1b35977a22138d469b46fe76aa7859a5739a15f44610f4a8fc). 

But I think it makes no sense to correlate two clients together, so I would delete it anyway.